pkg/pillar/cmd/README.md

```diff
@@ -0,0 +1,273 @@
+# Agent Services
+
+This directory contains agent services for pillar.
+
+Each subdirectory contains exactly one agent, whose name is the name of the directory.
+
+Look at [verifier](./verifier/) for an example of the implementation.
+
+## Design Pattern
+
+The design pattern for each agent is as follows. When creating a new agent, ensure
+that you follow the design pattern.
+
+The package name in the agent's directory is the same as the directory name.
+It should contain at least the following files:
+
+* `README.md` describing the agent and its purpose.
+* `run.go` containing the main library entry point for the agent.
+
+In addition, it should contain the following subdirectories:
+
+* `lib/` contains core logic, has no dependency on being called command-line or pubsub, unless pubsub is part of the core business logic. It is important to keep these functions as simple and straightforward as possible. Unit tests of business logic should be here as well.
```

Comment:
What if microservice is bigger and consists of multiple components (each implemented as a separate package)?
For example, [DPC Reconciler](https://github.com/lf-edge/eve/tree/master/pkg/pillar/dpcreconciler) is a component of the NIM microservice. Does it mean that it should be moved to `pillar/cmd/nim/lib/dpcreconciler`? I'm afraid it would lead to a very deeply nested directory structure.

Reply:
Why should it move? As it stands now, `dpcreconciler` is a library that is pulled in by NIM. Why should it have to move? The focus here is entirely on the entrypoints for agents and their structure.

Using your specific example, I would see nothing changing for `dpcreconciler`, while NIM would (eventually) move to a `lib/`, `pubsub/`, `cmd/` layout.